lib/hawkular/base_client.rb

```diff
@@ -174,14 +174,16 @@ def tenant_header
     end
 
     def handle_fault(f)
+      http_code = (f.respond_to?(:http_code) ? f.http_code : 0)
+      fail HawkularException.new('Invalid credentials', http_code) if f.instance_of? RestClient::Unauthorized
```

Comment:
Yeah, it should be, but the problem is that the REST API returns HTML response for this cases, (this is the problem that this PR is trying to address), one thing I can do is try to parse as a JSON but if it fails I can return a generic "Unauthorized Access" message.